src/functions/module_filter.ts

```diff
@@ -1,9 +1,95 @@
+import { cellsSortingCriteria } from "../helpers/sort";
 import { _t } from "../translation";
-import { AddFunctionDescription, Arg, isMatrix, Matrix, Maybe, ValueAndFormat } from "../types";
+import {
+  AddFunctionDescription,
+  Arg,
+  CellValue,
+  CellValueType,
+  isMatrix,
+  Locale,
+  Matrix,
+  Maybe,
+  ValueAndFormat,
+} from "../types";
 import { NotAvailableError } from "../types/errors";
 import { arg } from "./arguments";
 import { assertSameDimensions, assertSingleColOrRow } from "./helper_assert";
-import { assert, matrixMap, toBoolean, toMatrix, transposeMatrix } from "./helpers";
+import { toScalar } from "./helper_matrices";
+import {
+  assert,
+  matrixMap,
+  toBoolean,
+  toMatrix,
+  toNumber,
+  toString,
+  transposeMatrix,
+} from "./helpers";
+
+function sortMatrix(
+  _range: Matrix<ValueAndFormat>,
+  locale: Locale,
+  ...values: (ValueAndFormat | Matrix<ValueAndFormat>)[]
+): Matrix<ValueAndFormat> {
+  const criteriaIndex: (0 | 1)[] = [];
+  const sortColumns: Matrix<CellValue> = [];
+  const nRows = _range.length;
+  for (let i = 0; i < values.length; i += 2) {
+    criteriaIndex.push(toBoolean(toScalar(values[i + 1]).value) ? 1 : 0);
```

Comment:
that's probably the way it's supposed to work in other spreadsheets, but I don't understand why we are only checking the first value in the range (by using `toScalar`)

Reply:
Because here we are looking for the booleans "isAscending", so we need a scalar value